### ISSUE TITLE
roslib does not provide ros::package::getPath equivalents for catkin

### ISSUE THREAD

**Issue by trainman419** [ros/ros#65]
In rosbuild, it was possible to use rosbuild_download_data to download data into the source directory, and then get the path to that data with `ros::package::getPath()`.

In catkin, it's no longer recommended to download data into the source directory, so we download it into the package's share directory instead. Unfortunately, there isn't a matching ros::package function to get the share path for our package, so it's much more difficult to find files downloaded there.

tl;dr: please add `ros::package::getSharePath()` or similar to roslib

**Comment by dirk-thomas** [comment]
`roslib` uses `rospack` to resolve resources but since `rospack` was designed for `rosbuild` workspaces it is not aware of the devel space introduced with `catkin`. Therefore I don't see a way to implement the requested function `ros::package::getSharePath()` in a way which works for devel- as well as install space.

The only tool currently able to resolve this is `catkin_find` (which is a Python script). The path you would like to have can be fetched by calling:

```
catkin_find PKGNAME --share --first-only
```

`catkin_find` is provided by `catkin_pkg` and there is currently no C++ equivalent to that functionality.

**Comment by dirk-thomas** [comment]
In cases where you are using the downloaded data for tests you could work around this limitation by embedding the location information into the executable (e.g. passing it as a definition into the test: `-DMYTEST_DATA_DIR=${CATKIN_DEVEL_PREFIX}/share/${PROJECT_NAME}`).

**Comment by dirk-thomas** [comment]
I will close this ticket with the label "wontfix" since `roslib` does not contain any `catkin` specific knowledge. There is no other solution expected beside using `catkin_find`.
